app/presenters/vote_button_presenter.rb

```diff
@@ -1,9 +1,71 @@
 class VoteButtonPresenter < ApplicationPresenter
-  def vote?
-    parents_vote.blank?
+  presents :parent
+
+  def vote
+    @vote ||= Vote.find_by(user_id: h.current_user, voteable_id: parent)
+  end
+
+  def buttons
+    if !h.user_signed_in?
```

Comment:
I think I'd opt for a guard clause here since there's a lot more happening when the user is signed in. something like:

```ruby
return static_icons unless h.user_signed_in?

if vote.blank?
  ...
end
```

Reply:
The `case` might come in handy if we weren't using convenience methods here. I'm envisioning something like:

```ruby
case vote&.value
when 1
  up_vote_sequence
when -1
  down_vote_sequence
else
  new_vote_sequence
end
```